iiasa/message_ix
# Extend explanations and update signature of creating data frames in Westeros tutorials

This PR extends and updates the Westeros tutorials as well as extends the "Examples" section in the `make_df` documentation. PR #523 should be merged before merging this PR. After doing that the solution of the objective value of the Westeros baseline tutorial needs to be updated again to let the tutorial tests pass.
Closes #502. 

## Done in this PR
- Extended explanations incl. documentation for the utility function `make_df`
- Updated signature of creating data frames.
- Updated calculation of the `historical_new_capacity` and the corresponding tests 
- Applied code style `black`.
- Updated the quotation of variables and keywords.

## How to review

- Note that the CI checks all pass.
- Note the release notes were updated.
- Run `jupyter notebook` and go through the westeros baseline tutorial.
  - See if the explanations are correct/understandable, especially the historical activity and historical new capacity.
  - See if the quoting style of the different variables, keywords are readable as is.
  - See if new calculation of the `historical_new_capacity` is correct.
- Note the difference in the files of the `message_ix/testing` and `message_ix/tests` folder due to the updated calculation of `historical_new_capacity`. 
- Build the documentation and look at the [documentation for the utility function](https://docs.messageix.org/en/stable/api.html#module-message_ix.util) and the restructuring of the [tutorial overview page](https://docs.messageix.org/en/stable/tutorials.html#westeros-electrified).
- Possible, skim through the other tutorials.

## PR checklist
